livre/core/util/FrameUtils.cpp

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2015, EPFL/Blue Brain Project
+ *                     Jafet Villafranca Diaz <jafet.villafrancadiaz@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <livre/core/util/FrameUtils.h>
+
+namespace livre
+{
+
+FrameUtils::FrameUtils( const Vector2ui& frameRange,
+                        const Vector2ui& boundaries )
+    : _frameRange( INVALID_FRAME_RANGE )
+{
+    // frame range is outside the boundaries
+    if( frameRange[1] <= boundaries[0] || frameRange[0] >= boundaries[1] )
```

Comment:
is this correct ? You mean

if( frameRange[ 0 ] < bound[ 0 ] || frameRange[ 0 ] >= bound[ 1 ] ) ?